fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -1015,11 +1022,19 @@ public void rebuildIndex(@Nonnull FDBRecordStore store) {
     @Nonnull
     private CompletableFuture<TupleRange> buildEndpoints(@Nonnull FDBRecordStore store, @Nonnull RangeSet rangeSet,
                                                          @Nullable AtomicLong recordsScanned) {
+        boolean isIdempotent = store.getIndexMaintainer(index).isIdempotent();
+        final IsolationLevel isolationLevel =
+                isIdempotent ?
+                // if idempotent: since double indexing is harmless, we can skip the range conflict protection. At worse,
```

Comment:
It's a good place to illustrate on "range conflict protection": allow records being added to the range (so they may be or may not be seen by indexer) while scanning.